bad_connector_woocommerce/models/woo_backend/common.py

```diff
@@ -440,3 +498,31 @@ def cron_import_metadata(self, domain=None):
         """Cron for sync_metadata"""
         backend_ids = self.search(domain or [])
         backend_ids.sync_metadata()
+
+    def _domain_for_update_product_stock_qty(self):
+        """Domain to search WooCommerce product"""
+        return [
+            ("backend_id", "in", self.ids),
+            ("detailed_type", "=", "product"),
+            ("stock_management", "=", True),
+            ("backend_stock_manage", "=", True),
+        ]
+
+    def update_product_stock_qty(self):
```

Comment:
def update_product_stock_qty(self):
    """
    Export the Stock Inventory for WooCommerce products in Odoo.
    Update stock quantities for both individual products and product templates.
    """
    woo_product_obj = self.env["woo.product.product"]
    woo_product_tmpl_obj = self.env["woo.product.template"]
    domain = self._domain_for_update_product_stock_qty()
    woo_products = woo_product_obj.search(domain)
    woo_products.recompute_woo_qty()
    woo_products_templates = woo_product_tmpl_obj.search(domain)
    woo_products_templates.recompute_woo_qty()
    return True
    
   Note:- Kindly use the common method of recompute_woo_qty() by putting this method in one place